addons/barcodes/static/src/js/barcode_events.js

```diff
@@ -0,0 +1,148 @@
+odoo.define('web.BarcodeEvents', function(require) {
+"use strict";
+
+var core = require('web.core');
+var mixins = core.mixins;
+
+var BarcodeEvents = core.Class.extend(mixins.PropertiesMixin, {
+    timeout: null,
+    key_pressed: {},
+    buffered_key_events: [],
+    // Regexp to match a barcode input and extract its payload
+    // Note: to build in init() if prefix/suffix can be configured
+    regexp: /(.{3,})[\n\r\t]+/,
```

Comment:
Good to know, but I don't know how to do since when 'regexp' is defined, 'this' is still undefined (it could go in init but I find it clearer as a literal attribute).

I used a regexp because I find it intuitive. Just call match and you either get null if the string isn't a barcode or an array whose 2nd element is the barcode payload.